api/envoy/config/filter/listener/original_src/v2alpha1/original_src.proto

```diff
@@ -20,8 +20,9 @@ message OriginalSrc {
   // [#not-implemented-warn:]
   bool bind_port = 1;
 
-  // set the SO_MARK option on the upstream connection's socket. Used to ensure that non-local
-  // addresses may be routed back through envoy when binding to the original source address.
-  // [#not-implemented-warn:]
+  // Sets the SO_MARK option on the upstream connection's socket to the provided value. Used to
+  // ensure that non-local addresses may be routed back through envoy when binding to the original
+  // source address.
+  // [#proto-status: experimental]
```

Comment:
I realize this is not this PR, but does a mark of 0 make sense? Shouldn't this be an optional uint32 WKT wrapper? If a mark of 0 means, no mark, and this is OK, can you document that?

Reply:
A mark of 0 is equivalent to not marking it at all. I could document that for sure, though if I were to do that, I would probably want to make the mark option not set at all if it's zero. I can definitely do that as part of this PR. I'll update the docs while I'm at it too. Now that they're merged, it's not a big deal. :)

Reply:
OK thank you. Either option is fine with me.